doc/semimaxplus.xml

```diff
@@ -0,0 +1,101 @@
+#############################################################################
+##
+#W  semimaxplus.xml
+#Y  Copyright (C) 2016                                  Stuart A. Burrell
+##
+##  Licensing information can be found in the README file of this package.
+##
+#############################################################################
+##
+
+<#GAPDoc Label="IsXMatrixSemigroup">
+  <ManSection Label = "IsXMatrix">
+  <Heading>Matrix filters</Heading>
+    <Filt Name = "IsBooleanMatSemigroup"              Arg = "obj" Type = "Category"/>
+    <Filt Name = "IsMaxPlusMatrixSemigroup"           Arg = "obj" Type = "Category"/>
+    <Filt Name = "IsMinPlusMatrixSemigroup"           Arg = "obj" Type = "Category"/>
+    <Filt Name = "IsTropicalMatrixSemigroup"          Arg = "obj" Type = "Category"/>
+    <Filt Name = "IsTropicalMaxPlusMatrixSemigroup"   Arg = "obj" Type = "Category"/>
+    <Filt Name = "IsTropicalMinPlusMatrixSemigroup"   Arg = "obj" Type = "Category"/>
+    <Filt Name = "IsProjectiveMaxPlusMatrixSemigroup" Arg = "obj" Type = "Category"/>
+    <Filt Name = "IsNTPMatrixSemigroup"               Arg = "obj" Type = "Category"/>
+    <Filt Name = "IsIntegerMatrixSemigroup"           Arg = "obj" Type = "Category"/>
+    <Returns><K>true</K> or <K>false</K>.</Returns>
+    <Description>
+      The above are the currently supported types of matrix semigroups.
+    </Description>
+  </ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="IsFinite">
+  <ManSection>
+    <Attr Name = "IsFinite" Arg = "S"/>
+    <Returns><K>true</K> or <K>false</K>. </Returns>
+    <Description>
+      If <A>S</A> is a max-plus or min-plus matrix semigroup (i.e. belongs to
+      the category <Ref Filt = "IsMaxPlusMatrixSemigroup"/>), then
+      <C>IsFinite</C> returns <K>true</K> if <A>S</A> is finite and 
+      <K>false</K> otherwise. This method is based on
+      <Cite Key="Gaubert1996aa"/> (max-plus) and <Cite Key="Simon1978aa"/>
+      (min-plus). Note, in the min-plus case, all entries must be non-negative.
```

Comment:
What does this mean exactly that all entries must be non-negative? That for the method to work the entries have to be non-negative, or for the semigroup to be finite? 

Could you add a sentence that says something to the effect of "this method does not involve enumerating the semigroup."